### ISSUE TITLE
Build error while targeting linux

### ISSUE THREAD

**Issue by tsathishkumar** [mozilla-iot/webthing-rust#9]
I have already opened an [issue](https://github.com/actix/actix-web/issues/337) in actix-web. Just wondering if you tried to build by targeting for linux.

**Comment by mrstegeman** [comment]
Yes, Travis CI builds this project on Linux. The key is that you have to have a recent version of OpenSSL installed. Check out the Travis CI build script for this project: https://github.com/mozilla-iot/webthing-rust/blob/master/travis.sh#L3

**Comment by tsathishkumar** [comment]
Are the all python related installation also for openssl? Because just running the script gives me the below warning and in turn results in the same build error,

```
engines/afalg/e_afalg.c:30:4: warning: #warning "AFALG ENGINE requires Kernel Headers >= 4.1.0" [-Wcpp]
 #  warning "AFALG ENGINE requires Kernel Headers >= 4.1.0"
    ^
engines/afalg/e_afalg.c:31:4: warning: #warning "Skipping Compilation of AFALG engine" [-Wcpp]
 #  warning "Skipping Compilation of AFALG engine"
    ^
Cannot find "SSL_CTX_set0_ctlog_store" in podpath: cannot find suitable replacement path, cannot resolve link
Cannot find "d2i_SCT_LIST" in podpath: cannot find suitable replacement path, cannot resolve link
Cannot find "d2i_SCT_LIST" in podpath: cannot find suitable replacement path, cannot resolve link
Cannot find "i2d_SCT_LIST" in podpath: cannot find suitable replacement path, cannot resolve link
```

**Comment by mrstegeman** [comment]
No, the Python packages are unrelated. They're used for the test script.

**Comment by mrstegeman** [comment]
What Linux distro are you trying to build on?

**Comment by tsathishkumar** [comment]
Not really sure, this is my travis config - https://github.com/tsathishkumar/MySController-rs/blob/master/.travis.yml

and this is the build config in travis,

```
{
  "os": "linux",
  "dist": "trusty",
  "rust": "1.26.2",
  "cache": "cargo",
  "group": "stable",
  "script": [
    "cargo build --release --target x86_64-unknown-linux-gnu",
    "cargo test",
    "cargo deb --no-strip"
  ],
  "language": "rust",
  "before_script": [
    "cargo install cargo-deb || echo 'cargo-deb already installed, ignoring failure'"
  ],
  "before_install": [
    "./install_dependencies.sh",
    "sudo apt-get install libudev-dev"
  ]
}
```

**Comment by tsathishkumar** [comment]
Clearing cargo cache and re-running the build seems to have fixed the issue (with actix-web v0.6.14). Not sure if the issue was due to cargo cache, or actix-web v0.6.13. Anyway closing the issue.

Off topic question: Why are you not publishing the crate to crates.io yet?

**Comment by mrstegeman** [comment]
@tsathishkumar I would love to publish to crates.io. However, the `mdns` dependency is pulled from GitHub, which crates.io doesn't allow.

I've been trying to get the `mdns` author to publish his package to crates.io, but he is also using some GitHub dependencies, as he had to fork _another_ crate. Upstreaming his changes to `dns-parser` seems to have stalled.

There is also `libmdns` on crates.io, which is a fork of `mdns`. However, the author of that crate is totally unresponsive. Both libraries suffer from the same bug in their mDNS implementation, so I'm waiting to see which (if either) accepts my PR.

Alternatively, I could pull the required crate into this one and publish the whole thing together, but I'd rather not do that.

**Comment by mrstegeman** [comment]
@tsathishkumar Good news, my PR finally got merged! See #10. I will push to crates.io shortly.

**Comment by tsathishkumar** [comment]
Great news 😄
